overlord/devicestate/devicemgr.go

```diff
@@ -139,13 +140,24 @@ func (m *DeviceManager) Stop() {
 	m.runner.Stop()
 }
 
+func useStaging() bool {
+	return os.Getenv("SNAPPY_USE_STAGING_STORE") == "1"
+}
+
+func deviceAPIBaseURL() string {
+	if useStaging() {
+		return "https://myapps.developer.staging.ubuntu.com/identity/api/v1/"
```

Comment:
Looks like we're starting to reproduce quite a bit of code and patterns of our snapd/store package here.

Can we avoid that by moving those interactions into store proper?